### ISSUE TITLE
h2 client does not reconnect

### ISSUE THREAD

**Issue by olix0r** [BuoyantIO/linkerd#785]
Repro:

Run linkerd with a config like:

``` yaml
- protocol: h2
  experimental: true
  baseDtab: |
    /h2/localhost:4142 => /$/inet/127.1/8888;
    /h2 => /srv;
  servers:
  - port: 4142
  identifier:
    kind: io.l5d.headerToken
    header: ":authority"
```

Run nghttpd

```
:; nghttpd --no-tls -v 8888                                                                                                              
IPv4: listen 0.0.0.0:8888
IPv6: listen :::8888
```

Perform request:

```
:; curl --http2 http://localhost:4142/
```

Kill nghttpd

Curl again (should fail)

Restart nghttpd

Curl again.  Should succeed, but fails with:

```
l5d-err: com.twitter.finagle.DroppedWriteException
```

**Comment by zackangelo** [comment]
The error appears to change when building against Finagle 6.40.0: 

```E 1126 20:38:00.044 THREAD680: service failure
com.twitter.finagle.ChannelClosedException: null at remote address: /127.0.0.1:9897 from service: $/inet/127.0.0.1/9897. Remote Info: Upstream Address: Not Available, Upstream Client Id: Not Available, Downstream Address: /127.0.0.1:9897, Downstream Client Id: $/inet/127.0.0.1/9897, Trace Id: fde0d31c44d52c0e.fde0d31c44d52c0e<:fde0d31c44d52c0e
Caused by: java.nio.channels.ClosedChannelException
	at io.netty.channel.AbstractChannel$AbstractUnsafe.write(...)(Unknown Source)```

**Comment by olix0r** [comment]
this was fixed by #826 and released in 0.8.4
